aws/scripts/py/cache_invalidation.py

```
@@ -0,0 +1,302 @@
+
+Performs blue-green cache invalidation for CloudFront distributions.
+Processes staging distributions first, then production distributions.
+"""
+
+import boto3
+import logging
+import os
+import sys
+import time
+from enum import Enum
+from typing import Dict, Any, List, Optional
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)